src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs

```diff
@@ -103,10 +103,12 @@ public TemplateCreator(IEngineEnvironmentSettings environmentSettings)
                 return new TemplateCreationResult(CreationResultStatus.NotFound, templateInfo.Name, LocalizableStrings.TemplateCreator_TemplateCreationResult_Error_CouldNotLoadTemplate);
             }
 
-            string? realName = name ?? fallbackName ?? template.DefaultName;
+            string? realName = name is not null ? name : templateInfo.PreferDefaultName ? templateInfo.DefaultName : null;
+            realName = realName is null ? fallbackName : realName;
```

Comment:
Second look on this logic - it doesn't look correct.
CLI always passes the name of current directory as the fallback name. 
Therefore according to this logic:
- if name is `null`
- and if `PreferDefaultName` is true
- and if `templateInfo.DefaultName` is null

`fallbackName` will be still used, whereas it shouldn't.

if looks like it should be changed as following
```
var finalName = null;

if (!string.IsNullOrEmpty(name))
{
    //name was specified via "name" parameter - use it
    finalName = name;
}
//use default name
else if (templateInfo.PreferDefaultName)
{
   if (string.IsNullOrEmpty(templateInfo.DefaultName)
   {
      return new TemplateCreationResult(CreationResultStatus.TemplateIssueDetected, template.Name, "The template authoring issue: the template has `PreferDefaultName` set to `true`, but `DefaultName` was not configured.")
   }
   finalName = templateInfo.DefaultName;
}
//use fallback name if available
else
{
   if (string.IsNullOrEmpty(fallbackName)
   {
        if (string.IsNullOrEmpty(templateInfo.DefaultName))
       {
           //error - nothing to use as the name
        }
        else
        {
            finalName = templateInfo.DefaultName;
        }
   }
   else
   {
          finalName = fallbackName;
   }
}
```
However, on the other hand `name` is not always used, so i have doubts if it should be mandatory. But since current logic implies it is mandatory, let's keep it as. 
CLI has only few cases where `fallbackName` will be null. We should also check what VS can send as `fallbackName`.